src/server/saved_objects/migrations/core/migrate_raw_docs.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * This file provides logic for migrating raw documents.
+ */
+
+import { RawDoc } from './call_cluster';
+import { SavedObjectDoc, TransformFn } from './document_migrator';
+
+/**
+ * Applies the specified migration function to every saved object document in the list
+ * of raw docs. Any raw docs that are not valid saved objects will simply be passed through.
```

Comment:
question: What's the use case for just passing raw invalid documents through?

Reply:
The idea was that if the index contained raw docs that don't have the expected type -> attributes behavior, we don't know why those docs are there. Presumably some admin or someone placed them there, and it might not be good to just fail to boot until those docs are deleted.